scripts/check_material_profiles.py

```diff
@@ -1,89 +1,158 @@
-        # parse the definition file
-        guid_dict = OrderedDict()
-
-        materials_dir = self.get_materials_dir(self._materials_dir)
-
-        # go through all the preset settings files
-        for _, _, filenames in os.walk(materials_dir):
-            for filename in filenames:
-                file_path = os.path.join(materials_dir, filename)
-                if not filename.endswith(".xml.fdm_material"):
-                    continue
-
-                with open(file_path, "r", encoding = "utf-8") as f:
-                    content = f.read()
-
-                guid = self._get_guid(content)
-                if guid not in guid_dict:
-                    guid_dict[guid] = []
-
-                item_list = guid_dict[guid]
-                item_list.append({"file_name": filename,
-                                  "file_path": file_path})
-            break
-
-        has_invalid_files = False
-        for guid, file_item_list in guid_dict.items():
-            if len(file_item_list) <= 1:
-                continue
-            has_invalid_files = True
-
-            if guid is not None:
-                print("-> The following files contain the same GUID [%s]:" % guid)
-            else:
-                print("-> The following files DO NOT contain any GUID:")
-            for file_item in file_item_list:
-                print("    -- [%s]" % file_item["file_name"])
-            print("-> PLEASE make sure to generate unique GUIDs for each material.")
-
-        return not has_invalid_files
+NAMESPACES = {
+    "um": "http://www.ultimaker.com/material",
+    "cura": "http://www.ultimaker.com/cura",
+}
 
 
-if __name__ == "__main__":
-    script_dir = os.path.dirname(os.path.realpath(__file__))
-    root_dir = os.path.abspath(os.path.join(script_dir, ".."))
+class MaterialProfile:
+
+    def __init__(self, document: etree.Element, filename: Optional[Path]) -> None:
+        self.document = document
+        self.filename = filename
+
+    @classmethod
+    def from_file(cls, p: Path) -> 'MaterialProfile':
+        with p.open("rb") as f:
+            return cls(etree.fromstring(f.read()), p)
+
+    def brand(self) -> Optional[str]:
+        """Get the material brand name from the given lxml.etree root node.
 
-    validator = MaterialProfilesValidator(root_dir)
-    is_everything_validate = validator.validate()
+        @returns None if the brand cannot be found, otherwise the brand text.
+        """
+        node = self.document.xpath("./um:metadata/um:name/um:brand", namespaces=NAMESPACES)
+        return node[0].text if node else None
 
-    ret_code = 0 if is_everything_validate else 1
-    sys.exit(ret_code)
+    def guid(self) -> Optional[str]:
+        node = self.document.xpath("./um:metadata/um:GUID", namespaces=NAMESPACES)
+        return node[0].text if node else None
+
+
+class MaterialProfilesValidator:
+    """Material Profile validator that validates against an XML Schema"""
+
+    class ValidationError(Exception):
```

Comment:
Using nested classes here scopes the `ValidationError` within the `MaterialProfilesValidator`. Since it's a single-file script, it helped a bit with grouping related code together. Catching it can be done using `except MaterialProfilesValidator.ValidationError`